src/rust/src/pkcs12.rs

```diff
@@ -297,29 +297,31 @@ fn cert_to_bag<'a>(
     })
 }
 
+struct KeySerializationEncryption<'a> {
+    password: String,
```

Comment:
What's the motivation to switch from `PyBackedBytes` to a String?

Reply:
My thought here was that `decode_encryption_algorithm`'s goal was to pull details about the requested encryption from Python into a more Rust friendly format. That way by the time we get to internal calls like `serialize_bags` we're no longer working with unprocessed Python input (with the exception of `mac_alogrithm` which would have involved greater refactoring).

The other advantage is that since we use the password for both `serialize_key_and_certificates` (encrypting ShroudedKeyBag) and `serialize_bags` (encrypting the ContentInfo), we can rely on the same KeySerializationEncryption object without pulling the string out of Python twice or passing an extra argument.

Reply:
`PyBackedBytes` has teh advantage that it avoids an allocation+copy, so it'd be preferrable to stick to it.

Reply:
Ah, I was under the impression the extra allocation was occurring regardless. But looking closer, I think you're right. I've corrected it.